include/boost/geometry/algorithms/detail/overlay/get_turn_info.hpp

```diff
@@ -142,6 +143,51 @@ struct base_turn_handler
         ctype const dy = get<1>(a) - get<1>(b);
         return dx * dx + dy * dy;
     }
+
+    template
+    <
+            std::size_t IndexP,
+            std::size_t IndexQ,
+            typename UniqueSubRange1,
+            typename UniqueSubRange2,
+            typename TurnInfo
+    >
+    static inline void both_collinear(UniqueSubRange1 const& range_p,
+            UniqueSubRange2 const& range_q,
+            std::size_t index_p, std::size_t index_q,
+            TurnInfo& ti)
+    {
+#if ! defined(BOOST_GEOMETRY_USE_RESCALING)
+        ti.operations[IndexP].remaining_distance = distance_measure(ti.point, range_p.at(index_p));
+        ti.operations[IndexQ].remaining_distance = distance_measure(ti.point, range_q.at(index_q));
+
+        // pk/q2 is considered as collinear, but there might be
+        // a tiny measurable difference. If so, use that.
+        // Calculate pk // qj-qk
+        typedef detail::distance_measure
+            <
+                typename select_coordinate_type
+                    <
+                        typename UniqueSubRange1::point_type,
+                        typename UniqueSubRange2::point_type
+                    >::type
+            > dm_type;
+
+        dm_type const dm = get_distance_measure(range_q.at(index_q - 1),
+            range_q.at(index_q), range_p.at(index_p));
+
+        if (! dm.is_zero())
+        {
+            // Not truely collinear, distinguish for union/intersection
+            // If p goes left (positive), take that for a union
+            ui_else_iu(dm.is_positive(), ti);
```

Comment:
>     * Indeed, changing the value (a bit) should not matter. I will come back to this.
The value itself is indeed not sensitive, it can be changed a bit without changing behaviour. Of course if you change it completely, things solved by this feature might start to fail.